auto_nag/scripts/component.py

```diff
@@ -69,17 +80,36 @@ def get_bz_params(self, date):
             "f9": "CP",
         }
 
+    def failure_callback(self, bugid):
+        self.to_cache.remove(int(bugid))
+
+    def terminate(self):
+        self.add_to_cache(self.to_cache)
+
     def get_bugs(self, date="today", bug_ids=[]):
-        # Retrieve bugs to analyze.
-        bugs = super().get_bugs("component", date=date, bug_ids=bug_ids)
-        if len(bugs) == 0:
+        # Retrieve bugs ids to analyze with only the needed fields, see
```

Comment:
We should do one single query where we get all the data, similarly to the other "normal" BZCleaner scripts

Reply:
The code is actually doing a single Bugzilla request, I will update the comment

Reply:
We should do it similarly to the other scripts I think, rather than manually with our custom code

Reply:
We do use `BzCleaner.get_bugs` which does the heavy job. We have to override `bughandler` because `BzCleaner.bughandler` is stripping some fields when processing the bugs: https://github.com/mozilla/relman-auto-nag/blob/master/auto_nag/bzcleaner.py#L232

Reply:
Which data is it stripping that we'd need?

Reply:
For component, it was returning only `id` and `summary`, I think we could ask for it if we implements `has_product_component` in the component script but the behavior of including the minimum amount of returning only a subset of what was asked in the query was quite confusing to debug.

Reply:
@calixteman can you help here?

Reply:
@Lothiraldan did you fix your problem here ?

Reply:
I'm getting the behavior I want with this code change but I think @marco-c was not comfortable with the previous code. Is it better now @marco-c ?

Reply:
I think this works for now, we should fix the BzCleaner bughandler behavior though to make it less confusing (if I'm asking for a field, I want that field :P)

Reply:
Can you file an issue about that?

Reply:
Done https://github.com/mozilla/relman-auto-nag/issues/773